### PR TITLE
Add `ClimateModeMenu`

### PR DESCRIPTION
**Pull request by xplato**
Screenshot 2023-09-07 at 11 23 30 AM

### CODE REVIEW

File: src/lib/ui/thermostat/ClimateModeMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import { ChevronDownIcon } from 'lib/icons/ChevronDown.js'
+import { OffIcon } from 'lib/icons/Off.js'
+import { ThermostatCoolIcon } from 'lib/icons/ThermostatCool.js'
+import { ThermostatHeatIcon } from 'lib/icons/ThermostatHeat.js'
+import { ThermostatHeatCoolIcon } from 'lib/icons/ThermostatHeatCool.js'
+import { Menu } from 'lib/ui/Menu/Menu.js'
+import { ThermoModeMenuOption } from 'lib/ui/thermostat/ThermoModeMenuOption.js'
+
+const modes = ['heat', 'cool', 'heatcool', 'off'] as const
```
Comment (razor-x) [inline src/lib/ui/thermostat/ClimateModeMenu.tsx:9]: Can we use the HvacModeSetting type from the sdk? Something like
```suggestion
const modes: HvacModeSetting[] = ['heat', 'cool', 'heat_cool', 'off'] as const
```

File: src/lib/ui/thermostat/ClimateModeMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          onClick={() => {
+            onChange(m)
+          }}
+        />
+      ))}
+    </Menu>
+  )
+}
+
+function getModeIcon(mode: Mode): JSX.Element {
```
Comment (razor-x) [inline src/lib/ui/thermostat/ClimateModeMenu.tsx:52]: This is written as a component function, so we can use normal JSX, e.g., `<ModeIcon />`

```suggestion
function ModeIcon(mode: Mode): JSX.Element {
```

File: src/lib/ui/thermostat/FanModeMenu.tsx [hunk 3]
```diff
@@ -1,11 +1,11 @@
-import { CheckBlackIcon } from 'lib/icons/CheckBlack.js'
 import { ChevronDownIcon } from 'lib/icons/ChevronDown.js'
 import { FanIcon } from 'lib/icons/Fan.js'
 import { FanOutlineIcon } from 'lib/icons/FanOutline.js'
 import { Menu } from 'lib/ui/Menu/Menu.js'
-import { MenuItem } from 'lib/ui/Menu/MenuItem.js'
+import { ThermoModeMenuOption } from 'lib/ui/thermostat/ThermoModeMenuOption.js'
 
-type Mode = 'auto' | 'on'
+const modes = ['auto', 'on'] as const
```
Comment (razor-x) [inline src/lib/ui/thermostat/FanModeMenu.tsx:7]: The SDK should have a type for this one too.

Comment (xplato) [reply src/lib/ui/thermostat/FanModeMenu.tsx:7]: I don't see any references to this in `models.ts`

**Comment by xplato** [comment]
> Checking the story, it renders with no value selected (and the options open off screen).

@razor-x I can add a default for the lack of initial value, but I can't seem to replicate the menu position issue (on the deployed story or locally). Do you have any additional context/logs by chance?

**Comment by razor-x** [comment]
Try on Firefox. I think it's the same issue as https://github.com/seamapi/react/issues/424

**Comment by xplato** [comment]
> Try on Firefox. I think it's the same issue as https://github.com/seamapi/react/issues/424

I can't replicate on Firefox but I can on Safari. If there's an outstanding issue for it, should that be handled separately?

**Comment by razor-x** [comment]
@xplato Sure. I'm going to mark that one higher priority now.

File: src/lib/ui/thermostat/FanModeMenu.tsx (hunk 3 above)
Comment (razor-x) [reply src/lib/ui/thermostat/FanModeMenu.tsx:7]: https://github.com/seamapi/javascript/pull/278

Comment (xplato) [reply src/lib/ui/thermostat/FanModeMenu.tsx]: Cool; used in [0c0e45e](https://github.com/seamapi/react/pull/442/commits/0c0e45e9a3e084a3e0d49b5d72b24aec28e184ff)

**Review by razor-x — APPROVED** [review]

**Pull request merged**